bpf/kmesh/ads/include/local_ratelimit.h

```diff
@@ -0,0 +1,199 @@
+/* SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause) */
+/* Copyright Authors of Kmesh */
+
+#ifndef __KMESH_LOCAL_RATE_LIMIT_H__
+#define __KMESH_LOCAL_RATE_LIMIT_H__
+
+#include "bpf_log.h"
+#include "bpf_common.h"
+#include "kmesh_common.h"
+#include "listener/listener.pb-c.h"
+
+struct ratelimit_key {
+    union {
+        struct {
+            __u32 ipv4;   /* Destination IPv4 address. */
+            __u32 port;   /* Destination port. */
+            __u32 family; /* Address family (e.g., AF_INET) */
+        } sk_skb;
+    } key;
+};
+
+struct ratelimit_value {
+    __u64 last_topup; /* Timestamp of the last token refill (nanoseconds) */
+    __u64 tokens;     /* Current number of available tokens */
+};
+
+struct {
+    __uint(type, BPF_MAP_TYPE_HASH);
+    __type(key, struct ratelimit_key);
+    __type(value, struct ratelimit_value);
+    __uint(max_entries, 1000);
```

Comment:
Pls define a macro.

Reply:
Change to MAP_SIZE_OF_LISTENER, because its size will not exceed the number of listeners. Each listener corresponds to at most one ratelimit entry.